Remotion/Web/Development.WebTesting/Configuration/Legacy/TestSiteLayoutConfigurationElement.cs

```diff
@@ -29,12 +29,14 @@ public class TestSiteLayoutConfigurationElement : ConfigurationElement, IWebTest
     private readonly ConfigurationProperty _rootPathProperty;
     private readonly ConfigurationProperty _resourcesProperty;
     private readonly ConfigurationProperty _processPathProperty;
+    private readonly ConfigurationProperty _processArgumentsProperty;
 
     public TestSiteLayoutConfigurationElement ()
     {
       _rootPathProperty = new ConfigurationProperty("rootPath", typeof(string), "", ConfigurationPropertyOptions.IsRequired);
       _resourcesProperty = new ConfigurationProperty("resources", typeof(TestSiteResourceConfigurationElementCollection));
       _processPathProperty = new ConfigurationProperty("processPath", typeof(string), null);
+      _processArgumentsProperty = new ConfigurationProperty("processArguments", typeof(string), null);
```

Comment:
When do we make processPath not-nullable in the callchain? Since later code requires a not-null value, there's got to be code for this somewhere...

Reply:
This happens in the hosting strategy as some strategies (e.g. IIS) don't need a process path. Others, like the `AspNetCoreHostingStrategy` do need a process path and will throw an exception if it is not there.